src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,145 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> headNode;
+    private Node<T> tailNode;
+    private int currentSize;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value, tailNode, null);
+        if (tailNode == null) {
+            headNode = newNode;
+        } else {
+            tailNode.next = newNode;
+        }
+        tailNode = newNode;
+        currentSize++;
     }
-
+    
     @Override
     public void add(T value, int index) {
+        if (index == currentSize) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        Node<T> findNode = findNodeByIndex(index);
+        Node<T> newNode = new Node<>(value, findNode.prev, findNode);
+        if (index == 0) {
+            headNode.prev = newNode;
+            headNode = newNode;
+        } else {
+            newNode.prev.next = newNode;
+        }
+        findNode.prev = newNode;
+        currentSize++;
     }
-
+    
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
-
+    
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return findNodeByIndex(index).item;
     }
-
+    
     @Override
     public T set(T value, int index) {
+        checkIndex(index);
+        Node<T> newNode = findNodeByIndex(index);
+        if (newNode != null) {
```

Comment:
Consider removing the check for `newNode != null`, since the method `findNodeByIndex(index)` always returns a valid node for the given index. This makes the if condition redundant.